PerformanceEfficiency/AppServiceAutoscalingSample/Readme.md

```diff
@@ -69,23 +69,77 @@ The duration is set to 5 minutes. This is amount of time that Autoscale engine w
+
+#### HttpQueueLength metric explained
 
-The duration is set to 1 minute. This is amount of time that Autoscale engine will look back for metrics. So in this case, 1 minute means that every time autoscale runs, it will query metrics for the past minute. The instance limits are:  max 5 instances, min 1 instance. and the cool down setting is set to 5 minutes; The cool down setting is the amount of time to wait after a scale operation before scaling again. In this case, since the cooldown is 5 minutes and a scale operation just occurred, Autoscale will not attempt to scale again until after 5 minutes. This is to allow the metrics to stabilize first. These settings may not be valid for a real scenario, but are intentionally set to easily reproduce the autoscale conditions.
+This metric represents the average number of HTTP requests that had to sit on the queue before being fulfilled. A high or increasing HTTP Queue length is a symptom of a plan under heavy load.
 
 #### SocketOutboundTimeWait explained
 
-This metric represents the number of TCP connections in TIME_WAIT state. The reason this metric can affect scalability is that one socket in a TCP connection that is shut down cleanly will stay in the TIME_WAIT state for period of 4 minutes. If many connections are being opened and closed quickly then socket's in TIME_WAIT may begin to accumulate on a system. There are a finite number of socket connections that can be established at one time and one of the things that limits this number is the number of available local ports. If too many sockets are in TIME_WAIT you will find it difficult to establish new outbound connections and you will need to tune your App Service plan scale out settings and prevent the system reaching the limits.
+The SocketOutboundTimeWait metric is another networking related metric available for in the App Service custom autoscaling configuration. This metric represents the number of TCP connections in TIME_WAIT state. The reason this metric can affect scalability is that one socket in a TCP connection that is shut down cleanly will stay in the TIME_WAIT state for period of 4 minutes. If many connections are being opened and closed quickly then socket's in TIME_WAIT may begin  to accumulate on a system. There are a finite number of socket connections that can be established at one time and one of the things that limits this number is the number of available local ports. If too many sockets are in TIME_WAIT you will find it difficult to establish new outbound connections and you will need to tune your App Service plan scale out settings and prevent the system reaching the limits.
```

Comment:
```suggestion
The SocketOutboundTimeWait metric is a networking metric available for in the App Service custom autoscaling configuration. This metric represents the number of TCP connections in TIME_WAIT state. This metric can affect scalability because a socket in a TCP connection that is shut down cleanly will stay in the TIME_WAIT state for period of 4 minutes. If many connections are being opened and closed quickly then socket's in TIME_WAIT may begin  to accumulate. There are a finite number of socket connections that can be established at one time and is impacted by the number of available local ports . If too many sockets are in TIME_WAIT you will find it difficult to establish new outbound connections and you will need to tune your App Service plan scale out settings and prevent the system reaching the limits.
```